Cache backend: Crash when a / character is written as the root dir

## Version
rclone v1.39
os/arch: linux/amd64
go version: go1.9.2

## Command
The commands that crashes with the cache backend works fine for plain or encrypted gdrive.

### Crashes (see log below):
rclone lsd gdrive_cached:/
rclone lsd gdrive_cached:/any_directory

### Lists directories successfully:
rclone lsd gdrive_cached:
rclone lsd gdrive_cached:any_directory
## Log

```
2017/12/26 12:29:45 DEBUG : Using config file from "/home/user/.config/rclone/rclone.conf"
2017/12/26 12:29:45 DEBUG : rclone: Version "v1.39" starting with parameters ["rclone" "-vv" "lsd" "gdrive_cached:/"]
2017/12/26 12:29:46 DEBUG : gdrive_cached: wrapped gdrive:media at root /
2017/12/26 12:29:46 INFO  : gdrive_cached: Cache DB path: /home/user/.cache/rclone/cache-backend/gdrive_cached.db
2017/12/26 12:29:46 INFO  : gdrive_cached: Cache chunk path: /home/user/.cache/rclone/cache-backend/gdrive_cached
2017/12/26 12:29:46 INFO  : gdrive_cached: Chunk Memory: true
2017/12/26 12:29:46 INFO  : gdrive_cached: Chunk Size: 5M
2017/12/26 12:29:46 INFO  : gdrive_cached: Chunk Total Size: 10G
2017/12/26 12:29:46 INFO  : gdrive_cached: Chunk Clean Interval: 1m0s
2017/12/26 12:29:46 INFO  : gdrive_cached: Workers: 4
2017/12/26 12:29:46 INFO  : gdrive_cached: File Age: 6h0m0s
2017/12/26 12:29:46 INFO  : gdrive_cached: Cache Writes: false
2017/12/26 12:29:46 INFO  : gdrive_cached:/: Modify window is 1ms
panic: runtime error: slice bounds out of range

goroutine 40 [running]:
github.com/ncw/rclone/cache.(*Directory).Remote(0xc4203d4e00, 0x4, 0x4)
        /tmp/nix-build-rclone-1.39.drv-0/go/src/github.com/ncw/rclone/cache/directory.go:98 +0x117
github.com/ncw/rclone/cache.(*Persistent).GetDirEntries.func1(0xc42046c9a0, 0xff8f40, 0xc42046c9a0)
        /tmp/nix-build-rclone-1.39.drv-0/go/src/github.com/ncw/rclone/cache/storage_persistent.go:204 +0x2d5
github.com/ncw/rclone/vendor/github.com/coreos/bbolt.(*DB).View(0xc4201663c0, 0xc420048ae0, 0x0, 0x0)
        /tmp/nix-build-rclone-1.39.drv-0/go/src/github.com/ncw/rclone/vendor/github.com/coreos/bbolt/db.go:680 +0x9a
github.com/ncw/rclone/cache.(*Persistent).GetDirEntries(0xc4203d6640, 0xc4203d4e00, 0x0, 0xc4203d4e00, 0xc420048b68, 0x41355c, 0xc420048b68)
        /tmp/nix-build-rclone-1.39.drv-0/go/src/github.com/ncw/rclone/cache/storage_persistent.go:182 +0x7d
github.com/ncw/rclone/cache.(*Fs).List(0xc420095860, 0x0, 0x0, 0x0, 0xc420033300, 0x0, 0x0, 0x0)
        /tmp/nix-build-rclone-1.39.drv-0/go/src/github.com/ncw/rclone/cache/cache.go:461 +0xb3
github.com/ncw/rclone/fs.ListDirSorted(0x15a2cc0, 0xc420095860, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /tmp/nix-build-rclone-1.39.drv-0/go/src/github.com/ncw/rclone/fs/operations.go:629 +0x63
github.com/ncw/rclone/fs.walk.func2(0xc4204a3d40, 0xc42006ab40, 0xff8b28, 0x15a2cc0, 0xc420095860, 0x0, 0xc4204a3d38, 0xc4204d1980, 0xc4204a3d50, 0xc42019d230, ...)
        /tmp/nix-build-rclone-1.39.drv-0/go/src/github.com/ncw/rclone/fs/walk.go:120 +0x1e3
created by github.com/ncw/rclone/fs.walk
        /tmp/nix-build-rclone-1.39.drv-0/go/src/github.com/ncw/rclone/fs/walk.go:112 +0x265
```

--- a/backend/cache/cache.go
+++ b/backend/cache/cache.go
@@ -175,13 +175,23 @@ type Fs struct {
 	cleanupChan      chan bool
 }
 
+// parseRootPath returns a cleaned root path and a nil error or "" and an error when the path is invalid
+func parseRootPath(path string) (string, error) {
+	return strings.Trim(path, "/"), nil
+}
+
 // NewFs constructs a Fs from the path, container:path
-func NewFs(name, rpath string) (fs.Fs, error) {
+func NewFs(name, rootPath string) (fs.Fs, error) {
 	remote := config.FileGet(name, "remote")
 	if strings.HasPrefix(remote, name+":") {
 		return nil, errors.New("can't point cache remote at itself - check the value of the remote setting")
 	}
-	rpath = strings.Trim(rpath, "/")
+
+	rpath, err := parseRootPath(rootPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to clean root path %q", rootPath)
+	}
+
 	remotePath := path.Join(remote, rpath)
 	wrappedFs, wrapErr := fs.NewFs(remotePath)
 	if wrapErr != nil && wrapErr != fs.ErrorIsFile {
@@ -200,7 +210,7 @@ func NewFs(name, rpath string) (fs.Fs, error) {
 	if *cacheChunkSize != DefCacheChunkSize {
 		chunkSizeString = *cacheChunkSize
 	}
-	err := chunkSize.Set(chunkSizeString)
+	err = chunkSize.Set(chunkSizeString)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to understand chunk size %v", chunkSizeString)
 	}

